### PR TITLE
feature/GE-827: Removed mixin, added middleware

### PR DESCRIPTION
**Pull request by AltusBarry**
No need for a subclass with mixin anymore. Should allow for theme templates for 3rd party apps to "just work".

### CODE REVIEW

File: authentication_service/middleware.py [hunk 1]
```diff
@@ -35,3 +35,33 @@ def process_response(self, request, response):
...
+                templates["new"].append({"name": name, "path": dir_path})
+                templates["original"].append(file_name)
+
+            joined_names = ",".join(templates["original"])
+            for template in templates["new"]:
+                prepend_list = []
+                if template["name"] not in joined_names:
+                    prepend_list.append("%s%s" % (
+                        template["path"], template["name"]))
+                response.template_name = prepend_list + response.template_name
```
Comment (cobusc) [inline authentication_service/middleware.py]: Is `response.template_name` a list?

**Comment by cobusc** [comment]
Awesome rework. +1

File: authentication_service/middleware.py (hunk 1 above)
Comment (AltusBarry) [reply authentication_service/middleware.py]: Yip, dunno why the name is singular. Its the culmination of the templates after `get_template_names` has run I believe.

File: authentication_service/middleware.py [hunk 2]
```diff
@@ -35,3 +35,33 @@ def process_response(self, request, response):
...
+    def process_template_response(self, request, response):
+        theme = request.GET.get("theme", None) or request.COOKIES.get(
+            self.cookie_key)
+        if theme:
+            response.set_cookie(
+                self.cookie_key, value=theme, httponly=True
+            )
+            templates = {"original": [], "new": []}
+            for path in response.template_name:
+                file_name = path[path.rindex("/")+1:]
```
[the remark this answers is not in the document]
Comment (AltusBarry) [reply authentication_service/middleware.py]: Seems like a better idea, will use.

**Pull request merged**